src/client/Microsoft.Identity.Client.Broker/RuntimeBroker.cs

```diff
@@ -257,7 +257,7 @@ private IntPtr GetParentWindow(CoreUIParent uiParent)
                 _logger))
             {
                 using (var readAccountResult = await s_lazyCore.Value.ReadAccountByIdAsync(
-                    acquireTokenSilentParameters.Account.HomeAccountId.ObjectId,
+                    authenticationRequestParameters.Account.HomeAccountId.ObjectId,
```

Comment:
So `acquireTokenSilentParameters.Account` ends up being the argument that was passed into ATS (null in case of login hint used); while `authenticationRequestParameters.Account` ends up being the account that MSAL finds in cache when login hint is used.

Reply:
If what peter said is true, should we be checking local cache for an account instead of going directly to WAM? We have decided to avoid checking local cache for tokens so wondering if it is the same for accounts with brokers?

Reply:
The logic of `AcquireTokenSilent(login_hint)` with WAM is: 

1. go to our cache and try to find `Account` object that matches `login_hint`. Fail with UI Required ex if not found or if 2+ are found.
2. call `AcquireTokenSilent(Account)`
3. look in our cache for AT 
4. look in our cache for RT (this is a bug, it is tracked). If found, refresh the RT.
5. go to WAM